### ISSUE TITLE
Importing rikaichamp's dictionary?

### ISSUE THREAD

**Issue by rivke41levp656** [FooSoft/yomichan#1599]
I want to import the dictionary that rikaichamp uses but am having trouble. It seems to use data from https://hikibiki.app, but there is no obvious way to download dictionary files from there. That site seems to store the data in sqlite files located in your firefox profile, but attempting to import these with `yomichan import` results in 
`Conversion process failed: no such table: dict`

**Comment by toasted-nutbread** [comment]
I haven't looked deeply into this, but check out https://github.com/birchill/hikibiki-data and see if there's anything meaningful in there.

**Comment by Thermospore** [comment]
At a glance, both rikaichamp and https://hikibiki.app just use JMdict. You can either

1. download a prebuilt copy of jmdict from the yomichan info page
2. use yomichan import to build your own copy of jmdict at your leisure (which is what I like to do; it's a very active project) 

There are a bunch of derivatives of that rikai yaddayadda extension, so I haven't kept track, but it might also include the KireiCake/RikaiCake dictionary. To be brutally honest, it's kinda trash. Even if it has something not in your other dictionaries, you are probably better off googling it or dropping it

**Comment by rivke41levp656** [comment]
I don't think that is the case. It is mostly JMDict, but there are differences. Rikaichamp generally has more detailed entries, and is much more comprehensive particularly w.r.t slang terms. For example, 
だんとつ : yomichan with my compiled JMDict does not read it, instead picks up only だんと
rikaichamp's entry: `断トツ [だんトツ, ダントツ] (n,adj-no) (uk,abbr,sl) far and away the best; the very best; cream of the crop; crème de la crème; decisive lead; commanding lead (from 断然 + トップ)
`
I don't know where the difference is coming from, but I don't think it is KireiCake, as yomichan with KireiCake produces the same result as JMDict in above example. Both my KireiCake and JMDict versions are probably old, but searching for だんとつ on https://www.edrdg.org/jmdictdb/cgi-bin/srchform.py?svc=jmdict&sid=　 (which I assume is the most recent version) produces no results, as does searching it on https://kireicake.com/rikaicake-online/.

**Comment by Thermospore** [comment]
> For example,
> だんとつ : yomichan with my compiled JMDict does not read it, instead picks up only だんと
> rikaichamp's entry: `断トツ [だんトツ, ダントツ] (n,adj-no) (uk,abbr,sl) far and away the best; the very best; cream of the crop; crème de la crème; decisive lead; commanding lead (from 断然 + トップ)`

![image](https://user-images.githubusercontent.com/35350819/114109488-d8c62700-9881-11eb-906a-9ed8eeca0927.png)

That's JMdict. JMdict doesn't consider `だんとつ` as a reading or form of that word, technically speaking. If you search `断トツ`, `だんトツ`, or `ダントツ` in your yomichan it should come up. You can enable some options to have yomichan check other formats automatically (highly recommended). Here are my settings. That option I marked with a red dot converts `だんとつ` to `ダントツ`, which successfully pulls up the entry. This is probably what Rikaichamp is doing

![image](https://user-images.githubusercontent.com/35350819/114110295-acaba580-9883-11eb-8a9e-07e65fafc1de.png)

> yomichan with KireiCake produces the same result as JMDict in above example

the "kirecake" on the yomichan info page is actually a frankenstein of the [upper RikaiCake table](https://kireicake.com/rikaicake-online/) and a very old copy of jmdict. I think it was ripped from rikaichan? imo it should either be removed entirely or replaced by a standalone version of RikaiCake not containing old jmdict

**Comment by Thermospore** [comment]
> (from 断然 + トップ)

you are right, those types of notes are missing from yomichan import's jmdict builds. I made an issue to suggest including them a while back #1165

Glad to see other people have interest in this as well!

**Comment by rivke41levp656** [comment]
OK. You're right about the hiragana-katakana conversion, I didn't consider that. So I guess the differences between dictionaries come from either discrepancies in JMDict version (Rikaichamp autoupdates as it is web-based), or the fact that yomichan excludes those parenthetical notes. I agree that they ought to be included.

**Comment by FooSoft** [comment]
This must be a different sqlite schema. yomichan-import currently only supports sqlite imports for Rikai-style databases, which probably differ in some insignificant way from what you are trying to load.
